realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -109,7 +118,11 @@ private RealmCache(RealmConfiguration config) {
             // Create a new local Realm instance
             BaseRealm realm;
 
+
             if (realmClass == Realm.class) {
+                if (configuration.getDurability() != SharedGroup.Durability.MEM_ONLY) {
```

Comment:
I would move this check to `inMemory()` in the builder instead and throw an exception there instead.